File: test-plugins/org.yakindu.sct.generator.cpp.test/test-gen/org/yakindu/sct/generator/cpp/test/InternalEventLifeCycleTest.java
/**
 * Copyright (c) 2013 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.generator.cpp.test;

import java.util.Collection;

import org.junit.Before;
import org.junit.runner.RunWith;
import org.yakindu.sct.generator.c.gtest.GTest;
import org.yakindu.sct.generator.c.gtest.GTestRunner;
import org.yakindu.sct.generator.c.gtest.GTestHelper;

@GTest(sourceFile = "gtests/InternalEventLifeCycleTest/InternalEventLifeCycleTest.cc", program = "gtests/InternalEventLifeCycleTest/InternalEventLifeCycle", model = "testmodels/SCTUnit/InternalEventLifeCycle.sct")
@RunWith(GTestRunner.class)
public class InternalEventLifeCycleTest {

	protected final GTestHelper helper = new GTestHelper(this) {

		@Override
		protected void getSourceFiles(Collection<String> files) {
			super.getSourceFiles(files);
			files.add(getFileName(getTestProgram()) + ".cpp");
		}
	};

	@Before
	public void setUp() {
		helper.generate();
		helper.compile();
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/InternalEventLifeCycleTest/InternalEventLifeCycleTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "InternalEventLifeCycle.h"

InternalEventLifeCycle handle;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		internalEventLifeCycle_init(&handle);
	}
};


TEST_F(StatemachineTest, InternalEventLifeCycleTest) {					
	internalEventLifeCycle_enter(&handle);
	EXPECT_TRUE(internalEventLifeCycle_isStateActive(&handle, InternalEventLifeCycle_r1_A));
	EXPECT_TRUE(internalEventLifeCycle_isStateActive(&handle, InternalEventLifeCycle_r2_C));
	internalEventLifeCycleIface_raise_e(&handle);
	internalEventLifeCycle_runCycle(&handle);
	EXPECT_TRUE(internalEventLifeCycle_isStateActive(&handle, InternalEventLifeCycle_r1_A));
	EXPECT_TRUE(internalEventLifeCycle_isStateActive(&handle, InternalEventLifeCycle_r2_D));
	internalEventLifeCycle_runCycle(&handle);
	EXPECT_TRUE(internalEventLifeCycle_isStateActive(&handle, InternalEventLifeCycle_r1_A));
	EXPECT_TRUE(internalEventLifeCycle_isStateActive(&handle, InternalEventLifeCycle_r2_D));
	internalEventLifeCycleIface_raise_f(&handle);
	internalEventLifeCycle_runCycle(&handle);
	EXPECT_TRUE(internalEventLifeCycle_isStateActive(&handle, InternalEventLifeCycle_r1_A));
	EXPECT_TRUE(internalEventLifeCycle_isStateActive(&handle, InternalEventLifeCycle_r2_C));
	internalEventLifeCycle_runCycle(&handle);
	EXPECT_TRUE(internalEventLifeCycle_isStateActive(&handle, InternalEventLifeCycle_r1_A));
	EXPECT_TRUE(internalEventLifeCycle_isStateActive(&handle, InternalEventLifeCycle_r2_C));
}


